internal/fnruntime/runner.go

```diff
@@ -201,36 +201,38 @@ type FunctionRunner struct {
 
 func (fr *FunctionRunner) Filter(input []*yaml.RNode) (output []*yaml.RNode, err error) {
 	pr := printer.FromContextOrDie(fr.ctx)
+	var fnOutput strings.Builder
 	if !fr.disableCLIOutput {
 		if fr.wasm {
-			pr.Printf("[RUNNING] WASM %q", fr.name)
+			fnOutput.WriteString(fmt.Sprintf("[RUNNING] WASM %q", fr.name))
 		} else {
-			pr.Printf("[RUNNING] %q", fr.name)
+			fnOutput.WriteString(fmt.Sprintf("[RUNNING] %q", fr.name))
 		}
 		if fr.displayResourceCount {
-			pr.Printf(" on %d resource(s)", len(input))
+			fnOutput.WriteString(fmt.Sprintf(" on %d resource(s)", len(input)))
 		}
-		pr.Printf("\n")
+		fnOutput.WriteString("\n")
 	}
 	t0 := time.Now()
 	output, err = fr.do(input)
 	if err != nil {
-		printOpt := printer.NewOpt()
-		pr.OptPrintf(printOpt, "[FAIL] %q in %v\n", fr.name, time.Since(t0).Truncate(time.Millisecond*100))
-		printFnResult(fr.ctx, fr.fnResult, printOpt)
+		fnOutput.WriteString(fmt.Sprintf("[FAIL] %q in %v\n", fr.name, time.Since(t0).Truncate(time.Millisecond*100)))
```

Comment:
Another thing I realized:
In case of `fn render` errors: `CaD UI` should use functionResult part of `fn render` response (resourcelist) to display the errors. It doesn't have to match the CLI output.